### PR TITLE
support `java` -> `nginx` -> `java` e2e test, and v3 protocol

### PR DESCRIPTION
**Pull request by mrproliu**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#4455 

___
### New feature or improvement
- add `java` -> `nginx` -> `java` e2e test.
- adapt the new v3 protocol backend.

### CODE REVIEW

**Comment by wu-sheng** [comment]
I think related to #4617 too?

**Comment by mrproliu** [comment]
> I think related to #4617 too?

yes, I lost that one.

File: test/e2e/e2e-test/docker/lua/docker-compose.yml [hunk 1]
```diff
@@ -42,7 +42,27 @@ services:
...
 
-  provider:
+  provider-entry:
+    build:
+      context: ../../../
+      dockerfile: e2e-test/docker/Dockerfile.provider
+    networks:
+      - e2e
+    expose:
+      - 9091
```
Comment (kezhenxu94) [inline test/e2e/e2e-test/docker/lua/docker-compose.yml:52]:
```suggestion
      - 9090
```

File: test/e2e/e2e-test/docker/lua/docker-compose.yml [hunk 2]
```diff
@@ -42,7 +42,27 @@ services:
...
+    environment:
+      - SW_AGENT_COLLECTOR_BACKEND_SERVICES=oap:11800
+      - SW_AGENT_NAME=e2e-service-entry-provider
+    depends_on:
+      oap:
+        condition: service_healthy
+    healthcheck:
+      test: ["CMD", "sh", "-c", "nc -z 127.0.0.1 9091"]
```
Comment (kezhenxu94) [inline test/e2e/e2e-test/docker/lua/docker-compose.yml:60]:
```suggestion
      test: ["CMD", "sh", "-c", "nc -z 127.0.0.1 9090"]
```

File: test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/LuaE2E.java [hunk 3]
```diff
@@ -77,17 +77,22 @@
     @ContainerHostAndPort(name = "ui", port = 8080)
     private HostAndPort swWebappHostPort;
 
+    @SuppressWarnings("unused")
+    @ContainerHostAndPort(name = "provider-entry", port = 8080)
```
Comment (kezhenxu94) [inline test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/LuaE2E.java:81]:
```suggestion
    @ContainerHostAndPort(name = "provider-entry", port = 9090)
```

Comment (mrproliu) [reply test/e2e/e2e-test/src/test/java/org/apache/skywalking/e2e/LuaE2E.java:81]: Thanks for your help

**Pull request merged**